src/Symfony/Bundle/SecurityBundle/Command/UserPasswordEncoderCommand.php

```diff
@@ -32,10 +35,10 @@ protected function configure()
     {
         $this
             ->setName('security:encode-password')
-            ->setDescription('Encode a password.')
+            ->setDescription('Encodes a password.')
             ->addArgument('password', InputArgument::OPTIONAL, 'Enter a password')
-            ->addArgument('user-class', InputArgument::OPTIONAL, 'Enter the user class configured to find the encoder you need.')
-            ->addArgument('salt', InputArgument::OPTIONAL, 'Enter the salt you want to use to encode your password.')
+            ->addArgument('user-class', InputArgument::OPTIONAL, 'Enter the user class configured to find the encoder you need.', 'Symfony\Component\Security\Core\User\User')
+            ->addOption('empty-salt', null, InputOption::VALUE_NONE, 'Do not generate a salt or let the encoder generate one.')
```

Comment:
My main concern of this command from a DX point of view is the management of the random salt. This is the behavior that I'd like to see:

  * Assume that users never want to provide the salt value explicitly.
  * Therefore, by default, the command manages the salt:
    * If it's Bcrypt, don't generate a salt.
    * If it's an encoder of any other type, generate a salt.
  * If the users want to use their own salts, allow to do that via an option (`--salt` ?)

So this would be the behavior of the command:

```bash
// encodes the password for the default Symfony User class and generates the salt randomly
// if we use bcrypt, the command doesn't generate a salt, but the user is not informed about that
$ php app/console security:encode-password PASSWORD

// encodes the password for our custom User class and generates the salt randomly
// if we use bcrypt, the command doesn't generate a salt, but the user is not informed about that
$ php app/console security:encode-password PASSWORD AppBundle\Entity\User

// encodes the password for the default Symfony User class and uses our custom salt
// if we use bcrypt, the command doesn't use this salt and it informs about it to the user
$ php app/console security:encode-password PASSWORD --salt="..."

// encodes the password for our custom User class and uses our custom salt
// if we use bcrypt, the command doesn't use this salt and it informs about it to the user
$ php app/console security:encode-password PASSWORD AppBundle\Entity\User --salt="..."
```